Cabal/src/Distribution/Simple/GHCJS.hs

```diff
@@ -1801,7 +1801,7 @@ pkgRoot verbosity lbi = pkgRoot'
       let ghcjsProg = fromMaybe (error "GHCJS.pkgRoot: no ghcjs program") $ lookupProgram ghcjsProgram (withPrograms lbi)
       in  fmap takeDirectory (getGlobalPackageDB verbosity ghcjsProg)
     pkgRoot' UserPackageDB = do
-      appDir <- getAppUserDataDirectory "ghcjs"
+      appDir <- getXdgDirectory XdgData "ghcjs"
```

Comment:
This is wrong. Here we look what `ghcjs` (or `ghc`) does. This is not up to Cabal to decide.

Reply:
Yeah, this was something I imported from #7188 and forgot to clean up (I did fix the similar problem in GHC.hs)...